.github/workflows/sync-repo-dev.yml

```diff
@@ -0,0 +1,68 @@
+name: sync-repo-dev
+
+on:
+  schedule:
+    - cron: "25 18 * * 0"
+    - cron: "45 */4 * * *"
+  workflow_dispatch:
+    inputs:
+      step:
+        description: "Select the step to run"
+        required: true
+        default: all
+        type: choice
+        options:
+          - all
+          - gpustack
+          - runner/runtime
+
+jobs:
+  sync-gpustack:
+    if: github.event.schedule == '45 */4 * * *' || github.event_name == 'workflow_dispatch' && (github.event.inputs.step == 'all' || github.event.inputs.step == 'gpustack')
```

Comment:
The check doesn’t look very clean. How about we separate it into two workflows and give them clearer names?